server/src/main/java/org/apache/druid/server/coordination/SegmentLoadDropHandler.java

```diff
@@ -57,7 +58,7 @@
  * Responsible for loading and dropping of segments by a process that can serve segments.
  */
 @ManageLifecycle
-public class SegmentLoadDropHandler implements DataSegmentChangeHandler
```

Comment:
What's the reason to remove the interface? 
I think this class is one of the main implementations of it, so if required, isn't it possible to modify the interface itself instead?

Reply:
It wouldn't make sense to modify the interface since a `DataSegmentChangeRequest` itself is not aware of the loading mode.

Having `SegmentLoadDropHandler` implement the handler interface doesn't seem to serve any real purpose, since we are already creating a separate handler inside `processRequest` method. That handler then calls the outside `add` or `remove` methods. So I decided to just get rid of that interface.

For now, this seemed the cleanest approach. We can add it back in the future, if needed.

Please let me know what you think.